docs/content/en/docs/how-tos/builders/_index.md

```diff
@@ -208,3 +209,69 @@ The following `build` section instructs Skaffold to build a
+Users can write a custom build script which must abide by the following contract for skaffold to work as expected:
+
+### Contract between Skaffold and Custom Build Script
+
+Skaffold will pass in the following environment variables to the custom build script:
+
+| Environment Variable         | Description           | Expectation  |
+| ------------- |-------------| -----|
+| $IMAGES     | An array of fully qualified image names, separated by spaces.  | The custom build script is expected to build an image and tag it with each image name in $IMAGES. Each image should also be pushed if `$PUSH_IMAGE=true`. | 
+| $PUSH_IMAGE      | Set to true if each image in `$IMAGES` is expected to exist in a remote registry. Set to false if each image in `$IMAGES` is expected to exist locally.      |   The custom build script will push each image in `$IMAGES` if `$PUSH_IMAGE=true` | 
+| $BUILD_CONTEXT  | An absolute path to the directory this artifact is meant to be built from. Specified by artifact `context` in the skaffold.yaml.      | None. | 
+| Local environment variables | The current state of the local environment (e.g. `$HOST`, `$PATH)`. Determined by the golang [os.Environ](https://golang.org/pkg/os/#Environ) function.| None. |
+
+As described above, the custom build script is expected to:
+
+1. Build and tag each image in `$IMAGES`
+2. Push each image in `$IMAGES` if `$PUSH_IMAGE=true`
+
+Once the build script has finished executing, skaffold will try to obtain the digest of the newly built image from a remote registry (if `$PUSH_IMAGE=true`) or the local daemon (if `$PUSH_IMAGE=false`).
+If skaffold fails to obtain the digest, it will error out.
+
+#### Additional Environment Variables
+
+Skaffold will pass in the following additional environment variables for the following builders:
+
+##### Local builder
+| Environment Variable         | Description           | Expectation  |
+| ------------- |-------------| -----|
+| Docker daemon environment variables     | Inform the custom builder of which docker daemon endpoint we are using. Allows custom build scripts to work with tools like Minikube.| None. | 
+
+
+### Configuration
+
+To use a custom build script, add a `custom` field to each artifact in the `build` section of the skaffold.yaml.
+Currently, this only works with the build type `local`. Supported schema for `custom` includes:
+
+
+{{< schema root="CustomArtifact" >}}
+
+
+`buildCommand` is *required* and points skaffold to the custom build script which will be executed to build the artifact.
+`dependencies` tells the skaffold file watcher which files should be watched to trigger rebuilds and file syncs.  Supported schema for `dependencies` includes:
+
+
+{{< schema root="CustomDependencies" >}}
+
+#### Custom Build Scripts and File Sync
+Syncable files must be specified in the `paths` section of `dependencies` so that the skaffold file watcher knows to watch them. 
```

Comment:
Wait this is incorrect, Sync is another struct per artifact level which sync files without triggering a re-built.
```
apiVersion: skaffold/v1beta9
kind: Config
build:
  artifacts:
  - image: gcr.io/k8s-skaffold/node-example
    context: backend
    sync:
      '*.js': .
```
`Paths` are files to watch which will trigger a re-built. 
In dev flow, runner determines if a file needs to be synced using the "sync" maps. 
dependencies.path determine "paths that need to trigger the re-build"

Reply:
After discussion, this should be changed to 
```suggestion
Syncable files must also be included in the `paths` section of `dependencies` along with existing `sync` section so that the skaffold file watcher knows to watch them. 
```

Reply:
Cool, thanks!